OnlineStore/OnlineStore.BusinessLogic/Services/CategoryService.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using OnlineStore.BusinessLogic.Interfaces;
+using OnlineStore.Data.Entities;
+using OnlineStore.Data.Interfaces;
+using OnlineStore.DTO.DTO;
+using OnlineStore.DTO.Models;
+
+namespace OnlineStore.BusinessLogic.Services
+{
+    internal class CategoryService : ICategoryService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+
+        public CategoryService(IUnitOfWork unitOfWork,
+            IMapper mapper)
+        {
+            _mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+            _unitOfWork = unitOfWork ?? throw new ArgumentNullException(nameof(unitOfWork));
+        }
+        public async Task<CategoryDto?> GetCategoryByIdAsync(Int32 categoryId)
+        {
+            if (categoryId < 1)
+            {
+                throw new ArgumentException(nameof(categoryId));
+            }
+
+            var category = await _unitOfWork.Categories.GetByIdAsync(categoryId);
+
```

Comment:
It's better to add checks after obtaining a category by id. If it's not found, throw an exception. This applies to all methods where a single object is retrieved